### PR TITLE
Added maintenance window for prod postgress

### PR DESCRIPTION
**Pull request by VenkataChalla554**
Terraform plan 
 

      + maintenance_window {
          + day_of_week  = 0
          + start_hour   = 3
          + start_minute = 0
        }

### CODE REVIEW

File: terraform/application/config/production/variables.tfvars.json [hunk 1]
```diff
@@ -7,5 +7,10 @@
   "enable_monitoring": true,
   "external_hostname": "apply-for-qts-in-england.education.gov.uk",
   "namespace": "tra-production",
-  "uploads_storage_environment_tag": "Production"
+  "uploads_storage_environment_tag": "Production",
+  "azure_maintenance_window": {
+    "day_of_week": 0,
+    "start_hour": 3,
+    "start_minute": 0
+  }
```
Comment (thomasleese) [inline terraform/application/config/production/variables.tfvars.json:15]: Can we keep these variables in alphabetical order to make them easier to find?

File: terraform/application/.terraform.lock.hcl
Comment (thomasleese) [inline terraform/application/.terraform.lock.hcl:1]: Could we upgrade these dependencies in a separate commit (or preferably a separate PR)? Then we can deploy those on their own first, and it keeps this PR related only to the maintenance window changes.

**Review by thomasleese — COMMENTED** [review]
Thanks for this, could you add a description to the commit message explaining why this time has been chosen as the maintenance window and what that means?

It looks like there's also a linting issue that needs fixing.

File: terraform/application/config/production/variables.tfvars.json (hunk 1 above)
Comment (VenkataChalla554) [reply terraform/application/config/production/variables.tfvars.json]: fixed

File: terraform/application/variables.tf [hunk 2]
```diff
@@ -17,6 +17,8 @@ variable "azure_resource_prefix" {
   type = string
 }
 
+variable "azure_maintenance_window" { default = null }
```
Comment (thomasleese) [inline terraform/application/variables.tf:20]:
```suggestion
variable "azure_maintenance_window" {
  type = object({
      day_of_week  = number
      start_hour   = number
      start_minute = number
  })
  default = null
}
```

Just to make sure we always pass this in correctly.

Comment (VenkataChalla554) [reply terraform/application/variables.tf]: ya it make sense , updated the codes.

File: terraform/application/databases.tf [hunk 3]
```diff
@@ -29,4 +29,6 @@ module "postgres" {
   use_azure               = var.deploy_azure_backing_services
   azure_enable_monitoring = var.enable_monitoring
   azure_extensions        = ["pg_stat_statements", "pgcrypto"]
+
+  azure_maintenance_window = var.azure_maintenance_window
```
Comment (thomasleese) [inline terraform/application/databases.tf:33]:
```suggestion
  use_azure                = var.deploy_azure_backing_services
  azure_enable_monitoring  = var.enable_monitoring
  azure_extensions         = ["pg_stat_statements", "pgcrypto"]
  azure_maintenance_window = var.azure_maintenance_window
```

Just one minor thing, could we keep all the azure variables together so they're easier to see as a group.

**Review by thomasleese — COMMENTED** [review]
Sorry just some comment left, otherwise it's ready to go!

File: terraform/application/databases.tf (hunk 3 above)
Comment (VenkataChalla554) [reply terraform/application/databases.tf]: This is fixed .

**Comment by VenkataChalla554** [comment]
[add prod postgres maintenance window](https://trello.com/c/4NPqv8MF/1903-add-prod-postgres-maintenance-window)

**Review by thomasleese — APPROVED** [review]

**Pull request merged**